ocs_ci/deployment/aws.py

```diff
@@ -134,18 +150,190 @@ def destroy_cluster(self, log_level="DEBUG"):
             log_level (str): log level openshift-installer (default: DEBUG)
         """
         super(AWSIPI, self).destroy_cluster(log_level)
+        self.destroy_volumes()
 
-        try:
-            # Retrieve cluster name and AWS region from metadata
-            cluster_name = self.ocp_deployment.metadata.get("clusterName")
-            # Find and delete volumes
-            volume_pattern = f"{cluster_name}*"
-            logger.debug(f"Finding volumes with pattern: {volume_pattern}")
-            volumes = self.aws.get_volumes_by_name_pattern(volume_pattern)
-            logger.debug(f"Found volumes: \n {volumes}")
-            for volume in volumes:
-                self.aws.detach_and_delete_volume(
-                    self.aws.ec2_resource.Volume(volume['id'])
+
+class AWSUPI(AWSBase):
+    """
+    A class to handle AWS UPI specific deployment
+    """
+    def __init__(self):
+        self.name = self.__class__.__name__
+        super(AWSUPI, self).__init__()
+
+    class OCPDeployment(BaseOCPDeployment):
+        def __init__(self):
+            super(AWSUPI.OCPDeployment, self).__init__()
+            self.upi_repo_path = os.path.join(
+                self.cluster_path, 'openshift-misc',
+            )
+
+            self.upi_script_path = os.path.join(
+                self.upi_repo_path,
+                f'v3-launch-templates/functionality-testing'
+                f'/aos-4_2/hosts/'
+            )
+
+        def deploy_prereq(self):
+            """
+            Overriding deploy_prereq from parent. Perform all necessary
+            prerequisites for AWSUPI here.
+            """
+            super(AWSUPI.OCPDeployment, self).deploy_prereq()
+
+            # setup necessary env variables
+            upi_env_vars = {
+                'INSTANCE_NAME_PREFIX': config.ENV_DATA['cluster_name'],
+                'AWS_REGION': config.ENV_DATA['region'],
+                'rhcos_ami': 'ami-06c85f9d106577272',
+                'route53_domain_name': config.ENV_DATA['base_domain'],
+                'vm_type_masters': config.ENV_DATA['master_instance_type'],
+                'vm_type_workers': config.ENV_DATA['worker_instance_type'],
+                'num_workers': "3"  # TODO: read in num_workers
+            }
+            for key, value in upi_env_vars.items():
+                os.environ[key] = value
+
+            # ensure environment variables have been set correctly
+            for key, value in upi_env_vars.items():
+                assert os.getenv(key) == value, f"{os.getenv(key)} != {value}"
+
+            # git clone repo from openshift-qe repo
+            clone_repo(
+                constants.OCP_QE_MISC_REPO, self.upi_repo_path
+            )
+
+            # create install-dir inside upi_script_path
+            relative_cluster_path = "../../../../../../"
+            os.symlink(
```

Comment:
We need to make sure we have data available in NFS share, cause if it will be sym-linked there we won't see them under URL we provide for kubeconfig for example.

Reply:
The repo is being cloned in the `cluster_path` which should be on the NFS share. The sym link is being used by the install script, but the true contents exist at the `cluster_path`